src/test/java/org/zalando/zmon/notifications/pagerduty/webhook/PagerDutyWebHookControllerTest.java

```diff
@@ -0,0 +1,131 @@
+package org.zalando.zmon.notifications.pagerduty.webhook;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Resources;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.zalando.zmon.notifications.HttpEventLogger;
+import org.zalando.zmon.notifications.config.NotificationServiceConfig;
+import org.zalando.zmon.notifications.oauth.TokenInfoService;
+import org.zalando.zmon.notifications.pagerduty.client.PagerDutyClient;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Optional;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.zalando.zmon.notifications.ZMonEventType.PAGE_ACKNOWLEDGED;
+import static org.zalando.zmon.notifications.pagerduty.client.ClientTestUtils.mockAlert;
+
+@RunWith(MockitoJUnitRunner.class)
```

Comment:
I admit that I'm not experienced in this field. I had some previous experience with Mockito and it allowed me to get the job done. If you can provide some comprehensive examples I could give it a shot.